lasy/utils/denoise.py

```diff
@@ -0,0 +1,103 @@
+from lasy.profiles.transverse.hermite_gaussian_profile import (
+    HermiteGaussianTransverseProfile,
+)
+from lasy.profiles.transverse.transverse_profile import TransverseProfile
+from lasy.utils.mode_decomposition import hermite_gauss_decomposition
+
+
+def denoise_transverse_hg(
+    transverse_profile,
+    wavelength,
+    resolution=0.2e-6,
+    lo=[-2e-4, -2e-4],
+    hi=[2e-4, 2e-4],
+    n_modes_x=2,
+    n_modes_y=2,
+):
+    """
+    Denoise the transverse profile by decomposing it into a set of Hermite-Gaussian modes.
+
+    The profiles are weighted according to mode coefficients and then added.
+
+    Parameters
+    ----------
+    transverse_profile : class instance
+        An instance of a class or sub-class of TransverseProfile. Or it could be
+        a path to an image file from which intensity data will be read.
+        Defines the transverse envelope of the laser.
+
+    wavelength : float (in meter)
+        Central wavelength at which the Hermite-Gauss beams are to be defined.
+
+    resolution : float
```

Comment:
Please specify here too when a parameter is optional. Also true for others below.